File: output/llama_4_maverick/avatar/direct_translation/python/java/codeforces_11_A.java

import java.util.Scanner;

public class codeforces_11_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int d = Integer.parseInt(input[1]);
        int[] x = new int[n];
        input = scanner.nextLine().split(" ");
        for (int i = 0; i < n; i++) {
            x[i] = Integer.parseInt(input[i]);
        }
        int m = 0;
        for (int a = 0; a < n - 1; a++) {
            if (x[a] >= x[a + 1]) {
                int q = (x[a] - x[a + 1]) / d + 1;
                m += q;
                x[a + 1] += q * d;
            }
        }
        System.out.println(m);
    }
}

File: dataset/avatar/Python/Code/codeforces_11_A.py
n , d = map ( int , input ( ).split ( ) )
x = [ ]
m = 0
q = 0
x = input ( ).split ( )
for i in range ( len ( x ) ) :
    x [ i ] = int ( x [ i ] )
for a in range ( len ( x ) - 1 ) :
    if x [ a ] >= x [ a + 1 ] :
        q = int ( ( x [ a ] - x [ a + 1 ] ) / d + 1 )
        m = m + q
        x [ a + 1 ] = x [ a + 1 ] + q * d
print ( m )

